client/src/language/diagnosticsSupport.ts

```diff
@@ -109,25 +115,91 @@ const getEmbeddedLanguageType = (uri: vscode.Uri): EmbeddedLanguageType | undefi
   return undefined
 }
 
+const checkIsIgnoredDiagnostic = async (
+  diagnostic: vscode.Diagnostic,
+  originalTextDocument: vscode.TextDocument,
+  adjustedRange: vscode.Range
+): Promise<boolean> => {
+  if (await checkIsAlwaysIgnoredDiagnostic(diagnostic)) {
+    return true
+  }
+  if (await checkIsIgnoredDiagnosticOnAnonymousFunctionFirstLine(diagnostic)) {
+    return true
+  }
+  if (await checkIsIgnoredDiagnosticOnPythonUndefinedVariable(diagnostic, originalTextDocument, adjustedRange)) {
+    return true
+  }
+  if (await checkIsIgnoredShellcheckSc2154(diagnostic, originalTextDocument, adjustedRange)) {
+    return true
+  }
+  return false
+}
+
 const checkHasSupportedSource = (diagnostic: vscode.Diagnostic): boolean => {
   return supportedSources.some(
     (supportedSource) => diagnostic.source !== undefined && diagnostic.source.includes(supportedSource)
   )
 }
 
-const checkIsIgnoredPylanceUndefinedVariable = async (
+const checkIsAlwaysIgnoredDiagnostic = async (
+  diagnostic: vscode.Diagnostic
+): Promise<boolean> => {
+  if (
+    hasSourceWithCode(diagnostic, 'Flake8', 'W391') || // blank line at end of file
+    hasSourceWithCode(diagnostic, 'Pylint', 'C0114:missing-module-docstring') ||
+    hasSourceWithCode(diagnostic, 'Pylint', 'C0116:missing-function-docstring') ||
+    hasSourceWithCode(diagnostic, 'Pylint', 'C0305:trailing-newlines') ||
+    hasSourceWithCode(diagnostic, 'Pylint', 'C0415:import-outside-toplevel') ||
+
+    // From here, diagnostics we'd rather not always ignore, but which would require a call to the language server to check for their context.
+    // We're concerned making too many requests might cause performance issues.
```

Comment:
Really nice to have commented on that :+1: 
Also it's clear those potential negatives are not very impactful (mostly whitespace styling)